plugins/woocommerce/includes/class-wc-payment-gateways.php

```diff
@@ -113,6 +117,153 @@ public function init() {
 		}
 
 		ksort( $this->payment_gateways );
+
+		add_action( 'wc_payment_gateways_initialized', array( $this, 'on_payment_gateways_initialized' ), 10, 1 );
+		/**
+		 * Hook that is called when the payment gateways have been initialized.
+		 *
+		 * @since 8.5.0
+		 */
+		do_action( 'wc_payment_gateways_initialized' );
```

Comment:
```suggestion
		 * @since 8.5.0
		 *
		 * @param WC_Payment_Gateways $wc_payment_gateways The current WC_Payment_Gateways instance.
		 */
		do_action( 'wc_payment_gateways_initialized', $this );
```

What are your thoughts on also passing the class instance? The chief reason would be convenience: if someone wanted to unhook our callbacks, they may do so by hooking in here first.

Second reason is that although access is generally through the static `::instance()` method, I don't think anything actually stops there from being multiple distinct instances.

Reply:
At first I didn't want to pass the instance so as to ensure people use and are aware of the `::instance()` method, but you make a good point: people can still make their own and providing it here will make it more likely that they don't. ☺️